state model_state
control model_control
---
state updated_state
